pkg/metrics/service-monitor.go

```diff
@@ -59,6 +59,9 @@ func CreateServiceMonitors(config *rest.Config, ns string, services []*v1.Servic
 // based on the passed Service object.
 func GenerateServiceMonitor(s *v1.Service) *monitoringv1.ServiceMonitor {
 	labels := make(map[string]string)
+	if s == nil {
```

Comment:
https://github.com/operator-framework/operator-sdk/blob/b24f1b3dc397d738e53f4a5f4981391aa6f1a74a/pkg/metrics/service-monitor.go#L46-L48

How about handling this case earlier in CreateServiceMonitors where we can skip the loop if the service is nil:
```Go
for _, s := range services { 
 	if s == nil {
 	 	continue
 	}
 	sm := GenerateServiceMonitor(s) 
 	smc, err := mclient.ServiceMonitors(ns).Create(sm) 
```

I know `mclient.ServiceMonitors(ns).Create(sm) ` skips on nil, but I think it's more clear if we don't pass it in the first place.

Reply:
Done